test/src/unit/template.test.js

```diff
@@ -53,705 +55,519 @@ const mockRepos = {
-                description: 'The default set of templates for new projects in Codewind.',
-            };
-            before(() => {
-                fs.ensureDirSync(testWorkspaceConfigDir);
-                templateController = new Templates(testWorkspaceDir);
-                templateController.repositoryList = [...mockRepoList];
-                templateController.addProvider('valid Codewind repo', {
-                    getRepositories() {
-                        return [validCodewindRepo];
-                    },
+                after(() => {
+                    fs.removeSync(testWorkspaceDir);
                 });
-            });
-            after(() => {
-                fs.removeSync(testWorkspaceDir);
-            });
-            it(`updates the repository_list.json correctly`, async function() {
-                const expectedRepo = {
-                    ...validCodewindRepo,
-                    enabled: true,
-                    protected: true,
-                    name: 'Default templates',
-                    projectStyles: ['Codewind'],
-                };
-                await templateController.updateRepoListWithReposFromProviders();
-                const repoFile = fs.readJsonSync(templateController.repositoryFile);
-                repoFile.should.deep.equal([
-                    ...mockRepoList,
-                    expectedRepo,
-                ]);
-            });
-        });
-    });
-    describe('getReposFromProviders(providers)', function() {
-        const tests = {
-            'invalid provider: string': {
-                input: ['string'],
-                output: [],
-            },
-            'invalid provider: empty obj': {
-                input: [{}],
-                output: [],
-            },
-            'provider provides non-array': {
-                input: [{
-                    getRepositories() {
-                        return 'should be array';
-                    },
-                }],
-                output: [],
-            },
-            'provider provides a repo with URL': {
-                input: [{
-                    getRepositories() {
-                        return [{
-                            url: 'https://www.google.com/',
-                            description: 'not a GitHub repo',
-                        }];
-                    },
-                }],
-                output: [{
-                    url: 'https://www.google.com/',
-                    description: 'not a GitHub repo',
-                }],
-            },
-        };
-        for (const [testName, test] of Object.entries(tests)) {
-            describe(testName, function() {
-                it(`returns the expected repos`, async function() {
-                    const output = await Templates.getReposFromProviders(test.input);
-                    output.should.deep.equal(test.output);
+                it('returns the default templates and more', async function() {
```

Comment:
more is a bit vague ... could say default templates, and extra non-default templates